pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -823,14 +828,26 @@ func (c *Cluster) getConfigEnvVars(osdProps osdProperties, dataDir string) []v1.
 	return envVars
 }
 
-func (c *Cluster) provisionOSDContainer(osdProps osdProperties, copyBinariesMount v1.VolumeMount, provisionConfig *provisionConfig) v1.Container {
+func (c *Cluster) provisionOSDContainer(osdProps osdProperties, copyBinariesMount v1.VolumeMount, provisionConfig *provisionConfig) (v1.Container, error) {
 	osdOnPVC := osdProps.pvc.ClaimName != ""
 	osdOnPVCMetadata := osdProps.metadataPVC.ClaimName != ""
 	envVars := c.getConfigEnvVars(osdProps, k8sutil.DataDir)
 
 	// enable debug logging in the prepare job
 	envVars = append(envVars, setDebugLogLevelEnvVar(true))
 
+	// Ceph Drive Groups are specified
+	if len(osdProps.driveGroups) > 0 {
+		b, err := config.MarshalAsDriveGroupBlobs(osdProps.driveGroups)
+		if err != nil {
+			return v1.Container{}, errors.Wrap(err, "could not create OSD provisioning container")
+		}
+		// TODO: need to process host name, because ceph-volume won't ... should we use CRUSH
+		// host name (could be PVC name) or k8s node name? Should host names be processed at
+		// operator level or at provisioner level?
```

Comment:
Open question. Thoughts @travisn / @leseb ?

Reply:
Drive groups wouldn't be used with PVCs, right? I would expect the drive groups to be used with the CRUSH host name. The osd prepare job should already be detecting its CRUSH host name as well. Then the host name is returned to the operator from the prepare job so the operator knows what node selector to add to the osd deployment.

Reply:
After some reflection, I was thinking the answer is actually probably the kubernetes host name, because I think this should match what the orchestrator `host ls` command will show, yes?

Drive Groups won't be used for PVCs to my logic, and I also think the PVCs are what makes the CRUSH host name different from k8s host name, yes? So it shouldn't matter if we use CRUSH or k8s hostname here, but I think k8s hostname is the most correct based on my previous thought...

Reply:
Yes I would expect the `host ls` would match both the CRUSH and K8s names, where the k8s name is the `kubernetes.io/hostname` label and not the immutable node.Name.

Note that the PVC name is only used CRUSH map if `portable=true` on the storageClassDeviceSet. If `portable=false`, device sets would also use the host names.